core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/UpgradeImageInfoFactory.java

```diff
@@ -0,0 +1,37 @@
+package com.sequenceiq.cloudbreak.service.upgrade;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.cloud.model.Image;
+import com.sequenceiq.cloudbreak.core.CloudbreakImageCatalogException;
+import com.sequenceiq.cloudbreak.core.CloudbreakImageNotFoundException;
+import com.sequenceiq.cloudbreak.service.ComponentConfigProviderService;
+import com.sequenceiq.cloudbreak.service.image.ImageCatalogService;
+import com.sequenceiq.cloudbreak.service.image.StatedImage;
+
+@Service
+public class UpgradeImageInfoFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpgradeImageInfoFactory.class);
+
+    @Inject
+    private ComponentConfigProviderService componentConfigProviderService;
+
+    @Inject
+    private ImageCatalogService imageCatalogService;
+
+    public UpgradeImageInfo create(String imageId, Long stackId) throws CloudbreakImageNotFoundException, CloudbreakImageCatalogException {
```

Comment:
minor: I'd rename imageId parameter to targetImageId to avoid confusion

Reply:
Done.